src/app/shared/case-result-card/case-result-card/case-result-card.component.ts

```diff
@@ -0,0 +1,19 @@
+import { ChangeDetectionStrategy, Component, HostBinding, Input } from '@angular/core';
+import { Item } from '../../../model/item';
+import { RatingColorPipe } from '../../pipes/rating-color.pipe';
+
+@Component({
+  selector: 'app-case-result-card',
+  templateUrl: './case-result-card.component.html',
+  styleUrls: ['./case-result-card.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CaseResultCardComponent {
+  @Input() case: Item;
+
+  @HostBinding('class') get ratingColor(): string {
+    return this.case ? 'rating-color-bg--' + this.ratingColorPipe.transform(this.case.result_score) : '';
+  }
```

Comment:
Wouldn't it be easier to set the class from the parent component or is that not possible?

Reply:
@SandroGlueck  I think, I will leave it as it is. The case is given via async pipe into the component - integrating the class the same way will end up in messy code in the parent component 😊. Something like "'rating-color-bg--' + ((detailItem$ | async)?.result_score | ratingColor)".

The class is determined by the case properties itself, so IMO the component should be responsible for the default styling. If this does not fit, the class could be overwritten from the parent component anyway.